### PR TITLE
feat: Ignore specific errors for given pods

### PR DESCRIPTION
**Pull request by nikup**

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -57,6 +58,42 @@ func isIgnoredPod(name string) bool {
...
 }
 
+func isIgnoredErrorForPod(podName string, errorLog string) bool {
+	ignoredErrorsForPodNamePrefixesEnv := os.Getenv("IGNORED_ERRORS_FOR_POD_NAME_PREFIXES")
+	if ignoredErrorsForPodNamePrefixesEnv == "" {
+		return false
+	}
+
+	podErrorsMap := make(map[string][]interface{})
+	json.Unmarshal([]byte(ignoredErrorsForPodNamePrefixesEnv), &podErrorsMap)
```
Comment (ventsislav-georgiev) [inline helpers.go]: It is a good practice to always check the error returned and if != null log it.

File: helm/values.yaml [hunk 2]
```diff
@@ -18,6 +18,10 @@ ignoredNamespaces: ""
 # A comma-separated list of pod name prefixes to ignore
 ignoredPodNamePrefixes: ""
 
+# A map with specific errors to ignore from specific pods
+# "podName": ["error 1", "error 2"]
```
Comment (ventsislav-georgiev) [inline helm/values.yaml]: We can improve the example by having a valid json object here.
```
`{"podName":["error 1","error 2"]}`
```

File: helpers.go [hunk 3]
```diff
@@ -57,6 +58,42 @@ func isIgnoredPod(name string) bool {
...
+					return true
+				}
+			}
+		}
+	}
+
+	return false
+}
+
+func lastLogLine(logs string) string {
```
Comment (ventsislav-georgiev) [inline helpers.go]:
```suggestion
func lastNonEmptyLogLine(logs string) string {
```

**Review by ventsislav-georgiev — APPROVED** [review]
Looks good! Lets see if we can get it into the the parent repo?

File: helpers.go [hunk 4]
```diff
@@ -65,7 +65,11 @@ func isIgnoredErrorForPod(podName string, errorLog string) bool {
 	}
 
 	podErrorsMap := make(map[string][]interface{})
-	json.Unmarshal([]byte(ignoredErrorsForPodNamePrefixesEnv), &podErrorsMap)
+	err := json.Unmarshal([]byte(ignoredErrorsForPodNamePrefixesEnv), &podErrorsMap)
+	if err != nil {
+		klog.Infof("Failed to load IGNORED_ERRORS_FOR_POD_NAME_PREFIXES")
```
Comment (ventsislav-georgiev) [inline helpers.go]: Best to include the err in the log.

**Comment by nikup** [comment]
PR opened to the original repo from a different, cleaner branch: https://github.com/airwallex/k8s-pod-restart-info-collector/pull/39/

**Review by ventsislav-georgiev — APPROVED** [review]

**Pull request merged**